#!/usr/bin/env python
PACKAGE = "lidar_obstacle_detector"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("rotate_x",               double_t, 0, "Default: 0.0",    0.0, -10,  10)
gen.add("rotate_y",               double_t, 0, "Default: 0.0",    0.0, -10,  10)
gen.add("rotate_z",               double_t, 0, "Default: 0.0",    0.0, -10,  10)
gen.add("rotate_roll",            double_t, 0, "Default: 0.0",    0.0, -3.14,  3.14)
gen.add("rotate_pitch",           double_t, 0, "Default: 0.0",    0.0, -3.14,  3.14)
gen.add("rotate_yaw",             double_t, 0, "Default: 0.0",    0.0, -3.14,  3.14)

gen.add("use_pca_box",            bool_t,   0, "Default: True",   True)
gen.add("use_tracking",           bool_t,   0, "Default: True",   True)

gen.add("voxel_grid_size",        double_t, 0, "Default: 0.2",    0.2,  0.0,  1.0)

gen.add("roi_max_x",              double_t, 0, "Default: 70",     70,   0,    100)
gen.add("roi_max_y",              double_t, 0, "Default: 30",     30,   0,    100)
gen.add("roi_max_z",              double_t, 0, "Default: 1",      1,    0,    5)

gen.add("roi_min_x",              double_t, 0, "Default: -30",    -30,  -100, 0)
gen.add("roi_min_y",              double_t, 0, "Default: -30",    -30,  -100, 0)
gen.add("roi_min_z",              double_t, 0, "Default: -2.5",   -2.5, -5,   0)

type_enum = gen.enum([gen.const("RANSAC", str_t, "RANSAC", "Use RANSAC"),
                      gen.const("RGPF",   str_t, "RGPF",   "Use RGPF"),
                      gen.const("TRAVEL", str_t, "TRAVEL", "Use TRAVEL")],
                      "An enum to set ground segment type")
gen.add("ground_segment",         str_t,    0, "Select Ground Segment Type",   "RANSAC", edit_method=type_enum)

gen.add("ground_threshold",       double_t, 0, "Default: 0.3",    0.3,  0.0,  1.0)

gen.add("cluster_threshold",      double_t, 0, "Default: 0.6",    0.6,  0.0,  1.5)
gen.add("cluster_max_size",       int_t,    0, "Default: 5000",   5000, 0,    10000)
gen.add("cluster_min_size",       int_t,    0, "Default: 10",     10,   3,    100)

gen.add("displacement_threshold", double_t, 0, "Default: 1.0",    1.0,  0.0,  3.0)
gen.add("iou_threshold",          double_t, 0, "Default: 1.0",    1.0,  0.0,  1.0)

exit(gen.generate(PACKAGE, "obstacle_detector", "obstacle_detector_"))
